From b61fd7259c3949e222524440c8427b3dfb8c50aa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Thu, 12 May 2016 04:13:43 +0200
Subject: [PATCH] Fix unused variable compiler warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

lib/sys.c:643:19: error: 'file_types' defined but not used [-Werror=unused-const-variable=]
 static const char file_types[] = {
                        ^~~~~~~~~~

It is used only when debug/verbose is enabled.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 lib/sys.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/extras/mini-os/lib/sys.c b/extras/mini-os/lib/sys.c
index f8d7a7c..1420722 100644
--- a/extras/mini-os/lib/sys.c
+++ b/extras/mini-os/lib/sys.c
@@ -634,6 +634,7 @@ int closedir(DIR *dir)
 
 /* We assume that only the main thread calls select(). */
 
+#if defined(LIBC_DEBUG) || defined(LIBC_VERBOSE)
 static const char file_types[] = {
     [FTYPE_NONE]	= 'N',
     [FTYPE_CONSOLE]	= 'C',
@@ -646,6 +647,7 @@ static const char file_types[] = {
     [FTYPE_KBD]		= 'K',
     [FTYPE_FB]		= 'G',
 };
+#endif
 #ifdef LIBC_DEBUG
 static void dump_set(int nfds, fd_set *readfds, fd_set *writefds, fd_set *exceptfds, struct timeval *timeout)
 {
-- 
2.1.0

